tests/e2e/lifecycle/pvc_clone/test_clone_deletion_after_max_cluster_space_utilization.py

```diff
@@ -0,0 +1,209 @@
+        # Getting the total Storage capacity
+        self.ceph_cluster = CephCluster()
+        self.ceph_capacity = int(self.ceph_cluster.get_ceph_capacity())
+        logger.info(f"ceph_capacity: {self.ceph_capacity}")
+
+        # Getting the free Storage capacity
+        self.ceph_free_capacity = int(self.ceph_cluster.get_ceph_free_capacity())
+        logger.info(f"ceph_free_capacity: {self.ceph_free_capacity}")
+
+        # Change ceph full ratio
+        change_ceph_full_ratio(10)
+
+        # Use 10% of the free storage capacity in the test
+        self.capacity_to_use = int(self.ceph_capacity * 0.10)
+        logger.info(f"capacity_to_use: {self.capacity_to_use}")
+
+        self.num_of_clones = 200
+        # Calculating the PVC size in GiB
+        self.pvc_size = int(self.capacity_to_use / (self.num_of_clones + 2))
+        logger.info(f"pvc size: {self.pvc_size}")
+
+        self.pvc_obj = pvc_factory(
+            interface=interface_type, size=self.pvc_size, status=constants.STATUS_BOUND
+        )
+        self.pod_obj = pod_factory(
+            interface=interface_type, pvc=self.pvc_obj, status=constants.STATUS_RUNNING
+        )
+
+        # Calculating the file size as 70% of the PVC size - in MB
+        self.filesize = f"{int(self.pvc_size * 1024 * 0.7)}M"
+
+        self.pod_obj.run_io(
+            storage_type="fs",
+            size=self.filesize,
+            io_direction="write",
+        )
+
+        self.pod_obj.get_fio_results()
+        logger.info(f"IO finished on pod {self.pod_obj.name}")
+
+    def verify_alerts_via_prometheus(self, expected_alerts, threading_lock):
+        """
+        Verify Alerts on prometheus
+
+        Args:
+            expected_alerts (list): list of alert names
+            threading_lock (threading.Rlock): Lock object to prevent simultaneous calls to 'oc'
+
+        Returns:
+            bool: True if expected_alerts exist, False otherwise
+
+        """
+        prometheus = PrometheusAPI(threading_lock=threading_lock)
+        logger.info("Logging of all prometheus alerts started")
+        alerts_response = prometheus.get(
+            "alerts", payload={"silenced": False, "inhibited": False}
+        )
+        actual_alerts = list()
+        for alert in alerts_response.json().get("data").get("alerts"):
+            actual_alerts.append(alert.get("labels").get("alertname"))
+        for expected_alert in expected_alerts:
+            if expected_alert not in actual_alerts:
+                logger.error(
+                    f"{expected_alert} alert does not exist in alerts list."
+                    f"The actaul alerts: {actual_alerts}"
+                )
+                return False
+        return True
+
+    def test_clone_deletion_after_max_cluster_space_utilization(
+        self, interface_type, pvc_clone_factory, threading_lock
+    ):
+        """
+        Steps:
+            1. Have a cluster with OCP + ODF internal mode
```

Comment:
If the test cant run on external mode.. please add `@skipif_external_mode` decorator.

Reply:
sure